src/unistd.rs

```diff
@@ -2136,3 +2146,424 @@ mod setres {
+pub struct User {
+    /// username
+    pub name: String,
+    /// user password (probably encrypted)
+    pub passwd: CString,
+    /// user ID
+    pub uid: Uid,
+    /// group ID
+    pub gid: Gid,
+    #[cfg(not(target_os = "android"))]
+    /// user information
+    pub gecos: CString,
+    /// home directory
+    pub dir: PathBuf,
+    /// path to shell
+    pub shell: PathBuf,
+    #[cfg(not(any(target_os = "linux", target_os = "android")))]
+    /// login class
+    pub class: CString,
+    #[cfg(not(any(target_os = "linux", target_os = "android")))]
+    /// last password change
+    pub change: libc::time_t,
+    #[cfg(not(any(target_os = "linux", target_os = "android")))]
+    /// expiration time of account
+    pub expire: libc::time_t
+}
+
+impl From<*mut libc::passwd> for User {
+    fn from(pw: *mut libc::passwd) -> User {
+        unsafe {
+            User {
+                name: CStr::from_ptr((*pw).pw_name).to_string_lossy().into_owned(),
+                passwd: CString::new(CStr::from_ptr((*pw).pw_passwd).to_bytes()).unwrap(),
+                #[cfg(not(target_os = "android"))]
+                gecos: CString::new(CStr::from_ptr((*pw).pw_gecos).to_bytes()).unwrap(),
+                dir: PathBuf::from(OsStr::from_bytes(CStr::from_ptr((*pw).pw_dir).to_bytes())),
+                shell: PathBuf::from(OsStr::from_bytes(CStr::from_ptr((*pw).pw_shell).to_bytes())),
+                uid: Uid::from_raw((*pw).pw_uid),
+                gid: Gid::from_raw((*pw).pw_gid),
+                #[cfg(not(any(target_os = "linux", target_os = "android")))]
+                class: CString::new(CStr::from_ptr((*pw).pw_class).to_bytes()).unwrap(),
+                #[cfg(not(any(target_os = "linux", target_os = "android")))]
+                change: (*pw).pw_change,
+                #[cfg(not(any(target_os = "linux", target_os = "android")))]
+                expire: (*pw).pw_expire
+            }
+        }
+    }
+}
+
+/// Representation of a group, based on `libc::group`
+#[derive(Debug, Clone, PartialEq)]
+pub struct Group {
+    /// group name
+    pub name: String,
+    /// group ID
+    pub gid: Gid,
+    /// list of group members
+    pub mem: Vec<String>
+}
+
+impl From<*mut libc::group> for Group {
+    fn from(gr: *mut libc::group) -> Group {
+        unsafe {
+            Group {
+                name: CStr::from_ptr((*gr).gr_name).to_string_lossy().into_owned(),
+                gid: Gid::from_raw((*gr).gr_gid),
+                mem: Group::members((*gr).gr_mem)
+            }
+        }
+    }
+}
+
+impl Group {
+    unsafe fn members(mem: *mut *mut c_char) -> Vec<String> {
+        let mut ret = vec![];
+
+        for i in 0.. {
+            let u = mem.offset(i);
+            if (*u).is_null() {
+                break;
+            } else {
+                let s = CStr::from_ptr(*u).to_string_lossy().into_owned();
+                ret.push(s);
+            }
+        }
+
+        ret
+    }
+}
+
+#[derive(Debug)]
```

Comment:
These should always come after the doccomments.

Also is there a reason you didn't derive the other commonly-derived traits as well?

Reply:
moved below the doc-comments, and added the full set of `Debug, Copy, Clone, Eq, PartialEq, Hash, Default`, for `UserQuery`, `GroupQuery`, `Users`, `Groups` (where applicable, not all can be `Copy`, enums can't be `Default`)